METHOD_NAMING_PROPOSAL.md

```diff
@@ -0,0 +1,412 @@
+```
+
+</td>
+
+<td>
+
+```kotlin
+payPalLauncher = PayPalLauncher() { payPalBrowserSwitchResult ->
+  // Parse the result from the PayPal browser flow and return a nonce
+  payPalClient.onBrowserSwitchResult(payPalBrowserSwitchResult) 
+}
+
+// Retrieve data required to launch the PayPal flow in a browser
+payPalClient.tokenizePayPalAccount(context, request) { payPalResponse, error ->
+  // Launch the PayPal in a browser
+  payPalLauncher.launch(activity, payPalResponse) 
+}
+```
+
+</td>
+
+</tr>
+
+<tr>
+<th>v4 Local Payment</th>
+<th>v5 Local Payment</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to start local payment authorization
+// Or return result if no additional auth required
+localPaymentClient.startPayment(activity, request) { localPaymentResult, error ->
+  // Initiate browser flow and return nonce to listener
+  localPaymentClient.approveLocalPayment(activity, localPaymentResult)
+}
+```
+
+</td>
+
+<td>
+
+```kotlin
+localPaymentLauncher = LocalPaymentLauncher() { localPaymentBrowserSwitchResult ->
+  // Parse the result from the local payment browser flow and return a nonce
+  localPaymentClient.approveLocalPayment(localPaymentBrowserSwitchResult) 
+}
+
+// Retrieve data required to launch the local payment flow in a browser
+localPaymentClient.startPayment(context, request) { localPaymentResult, error ->
+  // Launch the local payment in a browser
+  localPaymentLauncher.launch(activity, localPaymentResult) 
+}
+```
+
+</td>
+
+</tr>
+<tr>
+<th>v4 3DS</th>
+<th>v5 3DS</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Retrieve data required to start 3DS authentication
+// Or return result if no additional auth required
+threeDSecureClient.performVerification(activity, request) { threeDSecureResult, error ->
+  // Launch 3DS flow in Cardinal SDK 
+  // Return nonce to listener
+  threeDSecureClient.continuePerformVerification(activity, request, threeDSecureResult)
+}
+```
+
+</td>
+
+<td>
+
+```kotlin
+threeDSecureLauncher = ThreeDSecureLauncher() { authenticationResult ->
+  // Parse the result from the 3DS auth flow and return a nonce
+  threeDSecureClient.continuePerformVerification(authenticationResult) 
+}
+
+// Retrieve data required to launch the 3DS flow
+// Instantiate Cardinal SDK
+threeDSecureClient.performVerification(context, request) { threeDSecureResult, error ->
+  // Launch 3DS flow in Cardinal SDK 
+  threeDSecureLauncher.launch(threeDSecureResult) 
+}
+```
+
+</td>
+
+</tr>
+
+<tr>
+<th>v4 Venmo</th>
+<th>v5 Venmo</th>
+</tr>
+<tr>
+
+<td>
+
+```kotlin 
+// Create Venmo payment context
+// Launch Venmo app and create nonce from payment context app switch result
+// Return nonce to listener
+venmoClient.tokenizeVenmoAccount(activity, request)
+```
+
+</td>
+
+<td>
+
+```kotlin
+venmoLauncher = VenmoLauncher(activity) { authChallengeResult ->
+  // Create nonce from payment context app switch result and vault if required
+  venmoClient.tokenizeVenmoAccount(authChallengeResult) 
```

Comment:
I like how this proposal keeps `tokenize()` as the last step in all of the flows!